iib/workers/tasks/utils.py

```diff
@@ -175,28 +176,49 @@ def _requires_max_ocp_version(bundle):
     return False
 
 
-def serve_index_registry(db_path):
+def serve_registry(base_dir, from_index=None, fbc_dir=None, db_path=None):
     """
     Locally start OPM registry service, which can be communicated with using gRPC queries.
 
     Due to IIB's paralellism, the service can run multiple times, which could lead to port
     binding conflicts. Resolution of port conflicts is handled in this function as well.
 
+    :param str base_dir: base directory to create temporary files in.
+    :param str from_index: index image to inspect.
+    :param str fbc_dir: path to file-based catalog directory.
     :param str db_path: path to index database containing the registry data.
     :return: tuple containing port number of the running service and the running Popen object.
     :rtype: (int, Popen)
     :raises IIBError: if all tried ports are in use, or the command failed for another reason.
     """
+    from iib.workers.tasks.build import _get_index_database
+
     conf = get_worker_config()
     port_start = conf['iib_grpc_start_port']
     port_end = port_start + conf['iib_grpc_max_port_tries']
 
+    assert (
+        len([i for i in (from_index, fbc_dir, db_path) if i]) == 1
+    ), 'Only one keyword parameter can be set'
+
+    if from_index:
+        if not is_image_fbc(from_index):
+            db_path = _get_index_database(from_index, base_dir)
+        else:
+            fbc_dir = get_index_fbc_dir(from_index, base_dir)
+
     for port in range(port_start, port_end):
+        if fbc_dir:
+            cmd = ['opm', 'serve', fbc_dir, '-p', str(port), '-t', '/dev/null']
+            cwd = os.path.abspath(os.path.join(fbc_dir, os.path.pardir))
```

Comment:
What does this variable mean? 'cwd'? Could you add a comment or change the name?

Reply:
cwd - current working directory - it is kwarg for https://docs.python.org/3/library/subprocess.html#subprocess.run 
That is why I would prefer to stick with this name.